src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -568,6 +580,44 @@ private static Details getDetailsForLambda(DetailAST ast) {
             return new Details(lcurly, rcurly, nextToken, shouldCheckLastRcurly);
         }
 
+        /**
+         * Collects details for Anonymous classes.
+         * @param ast a {@code DetailAST} value
+         * @return an object containing all details to make a validation
+         */
+        private static Details getDetailsForAnonymousClass(DetailAST ast) {
+            final Details details;
+            if (isAnonymousClassDef(ast)) {
+                final DetailAST objblockAST = ast.getLastChild();
+                final DetailAST lcurly = objblockAST.getFirstChild();
+                final DetailAST rcurly = objblockAST.getLastChild();
+                DetailAST nextToken = getNextToken(ast);
+                if (nextToken.getType() == TokenTypes.RPAREN) {
+                    nextToken = getNextToken(nextToken);
+                }
+                if (nextToken.getType() == TokenTypes.SEMI
+                        || nextToken.getType() == TokenTypes.COMMA) {
+                    nextToken = getNextToken(nextToken);
+                }
+                details = new Details(lcurly, rcurly, nextToken, false);
+            }
+            else {
+                details = new Details();
```

Comment:
@ps-sp , please explain for what case you need this ?
Details withtout details, looks strange.

Reply:
`LITERAL_NEW` not only represents anonymous classes but also object and array instantiations. Hence we first check if a `LITERAL_NEW` type of `AST` actually represents an anonymous class and if it does then we return a proper `Details` object.

If it doesn't then we return a dummy `Details` object without any details to avoid `NullPointerException` [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java#L173)

Reply:
Please remove default c-tor and lets use existing c-tor will all arguments explicitly defined.

> not only represents anonymous classes but also object and array 

So here is point, please name method exactly what it does `getDetailsForAnonymousClassOrObjectOrArray`

> ar = new int[4];

no details should be in this case only, please add it to UT input file that you created.

> new String[] {"br", }

this case should do have right curly, and we need to find it and put in Details. This is should be separate processing branch in IF-ELSE.
code should be like:
```
if (isAnonymous()) {
} else if (isArrayInitialization()) {
} else {
  // it is array creation without curly braces
 details = new Details(null, null, null, false);
}
```

> If it doesn't then we return a dummy Details object without any details to avoid NullPointerException here

All other methods getDetailsXXXXX also can return Detail object full of nulls inside, but they use the same c-tor. Please reuse it.

Reply:
>Please remove default c-tor and lets use existing c-tor will all arguments explicitly defined.

Done.

> So here is point, please name method exactly what it does 

Done.

> no details should be in this case only, please add it to UT input file that you created.

Added another test, `testBracePolicyAloneForArrayInit`. 

>this case should do have right curly, and we need to find it and put in Details. This is should be separate processing branch in IF-ELSE.

Done.